src/main/java/ca/warp7/frc2022/subsystems/Launcher.java

```diff
@@ -57,11 +72,10 @@ public void calcOutput() {
         if (targetRPS == 0.0)
             this.setVoltage(0.0);
         else
-            this.setVoltage((targetRPS + getError() * kLauncherKp) * kLauncherKv +
-                    kLauncherKs * Math.signum(targetRPS));
+            this.setVoltage(voltageCalculator.calculate(this.getError()));
```

Comment:
Calculate needs both the target and the current, otherwise you won't get the feed forward on the controller.  i.e.,

this.setVoltage(voltageCalculator( targetRPS, currentRPS ));

Reply:
Another thing I noticed was that the PIDController doesn't have upper and lower bounds set right now.

https://github.com/Team865/FRC-2022/blob/main/src/main/java/ca/warp7/frc2022/lib/control/PIDController.java#L15

If we set the motor using PercentOutput, then we should make sure that something guarantees that we never set the power on the motor outside a 0-1 value.  The motors are powering a flywheel;  we want to guarantee that the go in one direction (clockwise or counter clockwise, depending on what side you're looking at it from).

Edit - I would start by clamping it very low, like maybe 0-.05.  And having this code just feed in minimal values until we get the fly-wheel spinning properly.